letsencrypt-auto-source/pieces/bootstrappers/rpm_common_base.sh

```diff
@@ -0,0 +1,74 @@
+# If new packages are installed by BootstrapRpmCommonBase below, version
+# numbers in rpm_common.sh and rpm_python3.sh must be increased.
+
+FindInstallTool() {
+  if type dnf 2>/dev/null
+  then
+    tool=dnf
+  elif type yum 2>/dev/null
+  then
+    tool=yum
+
+  else
+    error "Neither yum nor dnf found. Aborting bootstrap!"
+    exit 1
+  fi
+}
+
+BootstrapRpmCommonBase() {
+  FindInstallTool
+  tool=$?
+
+  if [ "$ASSUME_YES" = 1 ]; then
+    yes_flag="-y"
+  fi
+  if [ "$QUIET" = 1 ]; then
+    QUIET_FLAG='--quiet'
+  fi
+
+  if ! $tool list *virtualenv >/dev/null 2>&1; then
+    echo "To use Certbot, packages from the EPEL repository need to be installed."
+    if ! $tool list epel-release >/dev/null 2>&1; then
+      error "Enable the EPEL repository and try running Certbot again."
+      exit 1
+    fi
+    if [ "$ASSUME_YES" = 1 ]; then
+      /bin/echo -n "Enabling the EPEL repository in 3 seconds..."
+      sleep 1s
+      /bin/echo -ne "\e[0K\rEnabling the EPEL repository in 2 seconds..."
+      sleep 1s
+      /bin/echo -e "\e[0K\rEnabling the EPEL repository in 1 seconds..."
+      sleep 1s
+    fi
+    if ! $tool install $yes_flag $QUIET_FLAG epel-release; then
+      error "Could not enable EPEL. Aborting bootstrap!"
+      exit 1
+    fi
+  fi
+
+  pkgs="
+    gcc
+    augeas-libs
+    openssl
+    openssl-devel
+    libffi-devel
+    redhat-rpm-config
+    ca-certificates
+  "
+
+  # Add the python packages
+  pkgs = "$pkgs
+    $1
```

Comment:
but the previous packages don't end in a newline?

or are you saying that we just don't actually care about newlines vs spaces and they're just there for programmer benefit?

Reply:
> but the previous packages don't end in a newline?

I think it does by putting the closing quote on a separate line from the last package. For example:
```
#!/bin/sh
a="
  foo
  bar
"
b="
  baz
  qux
"
c="
  $a
  $b
"
echo "$c"
```
outputs:
```

  
  foo
  bar

  
  baz
  qux


```
If `yum` doesn't care though, I wouldn't worry about it too much.

Reply:
Ah, yeah, yum definitely doesn't care because that's what we were doing before. Looking back now, I put it in on a newline to be precisely consistent with what it was before. Happy to fix this up while I'm here if it's more readable.

Reply:
If `yum` doesn't care and we were doing that before, it's fine with me.